presto-main/src/main/java/com/facebook/presto/sql/relational/UntranslatedExpression.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.relational;
+
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.sql.tree.Expression;
+
+import java.util.Objects;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * UntranslatedExpression is a RowExpression container holding an {@param Expression} object
+ * that cannot be translated directly while constructing PlanNode.
+ * Typical examples are cases with `WITH` clauses or alias that the type or arguments information
+ * are not directly available given the substrees are not formed yet.
+ * UntranslatedExpression should not exist after optimization or serialized over the wire.
+ * All UntranslatedExpression should be translated to other corresponding RowExpression objects ultimately.
+ */
+public final class UntranslatedExpression
```

Comment:
What about `RawExpression`? I feel that "untranslated" has too much emphasis on "translation" than necessary. Maybe it's just me.